### DISCUSSION TITLE
Feature Request: Personalized e-mail

### DISCUSSION THREAD

**Discussion by sagudelo1200** [pocketbase/pocketbase#269] [Q&A] [answered]
Hi, I apologize if this has been discussed before, I did not find references.

The verification/reset messages, etc are in English and the content of the message body cannot be changed.

I suggest that there is a way to enter custom texts in the UI, and that the links, app name, and other dynamic fields that are required, are handled with {{ variables }} e.g.

Ex:
Thank you for joining us at {{ APP_NAME }}
Click on the link below to verify your email address.
{{ VERIFY_EMAIL_LINK }}
Thank you,
{{ APP_NAME }} team

**Comment by ganigeorgiev** [comment]
Hi, this was discussed previously in https://github.com/pocketbase/pocketbase/discussions/57.

To summarize:
> Initially it was in the admin dashboard (see [earlier mockups](https://app.presentator.io/#/6mote1hf/prototypes/31143/screens/210839?mode=preview)) but I decided to replace it with event hooks because it provides more options for customizations and the behavior is more consistent when the dev uses PocketBase as framework (otherwise you may end up with something that seems configurable in the dashboard, but actually may not be used by the application due to some intercepted hook event).

I still haven't found a non ambiguous way to implement it in the UI and therefore, the current recommended way to customize the email templates is to use the Mailer hooks - https://pocketbase.io/docs/event-hooks/#mailer-hooks.

Suggestion how to better resolve the eventual "Framework vs UI" conflict are more than welcome.

Ideally sending emails and other post operations could be integrated into something like a drag&drop "Workflow builder/triggers" but that is a very big feature that will require a lot of time to design and implement properly, which I currently lack.

**Comment by ganigeorgiev** [comment, marked as answer]
@sagudelo1200 Just to let you know that support to customize the "default" user email templates from the Admin UI was added in [v0.5.0 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.5.0) as part of #275.

Developers are still able to override the defaults using the mailer hooks.

**Reply by sagudelo1200** [reply]
Thanks for that! It makes this excellent pocket application even better!

**Reply by sagudelo1200** [reply]
This was great!

We still need to be able to customize the confirmation messages for each of these actions.

**Reply by ganigeorgiev** [reply]
> We still need to be able to customize the confirmation messages for each of these actions.

What do you mean by "confirmation messages"?

If you mean the pages that are shown to the user after following the link in the email, you can provide your own custom one by changing the `ActionUrl` to something else (similar to how you probably would have a specific app login/register page).
